pkg/pfconfig/pfconfig.go

```diff
@@ -0,0 +1,225 @@
+	RSSITarget     float32            `json:"rssi_target"`
+	ChannelConfigs []LBTChannelConfig `json:"chan_cfg"`
+	RSSIOffset     float32            `json:"sx127x_rssi_offset"`
+}
+
+// LBTChannelConfig contains the listen-before-talk configuration for a channel.
+type LBTChannelConfig struct {
+	Frequency            uint64 `json:"freq_hz"`
+	ScanTimeMicroseconds uint32 `json:"scan_time_us"`
+}
+
+// RFConfig contains the configuration for one of the radios.
+type RFConfig struct {
+	Enable      bool    `json:"enable"`
+	Type        string  `json:"type"`
+	Frequency   uint64  `json:"freq"`
+	RSSIOffset  float32 `json:"rssi_offset"`
+	TxEnable    bool    `json:"tx_enable"`
+	TxFreqMin   uint64  `json:"tx_freq_min,omitempty"`
+	TxFreqMax   uint64  `json:"tx_freq_max,omitempty"`
+	TxNotchFreq uint64  `json:"tx_notch_freq,omitempty"`
+}
+
+// IFConfig contains the configuration for one of the channels.
+type IFConfig struct {
+	Description  string `json:"desc"`
+	Enable       bool   `json:"enable"`
+	Radio        uint8  `json:"radio"`
+	IFValue      int32  `json:"if"`
+	Bandwidth    uint32 `json:"bandwidth,omitempty"`
+	SpreadFactor uint8  `json:"spread_factor,omitempty"`
+	Datarate     uint32 `json:"datarate,omitempty"`
+}
+
+// MarshalJSON implements json.Marshaler
+func (c IFConfig) MarshalJSON() ([]byte, error) {
+	if !c.Enable {
+		return []byte(`{"desc": "disabled","enable": false}`), nil
+	}
+	return json.Marshal(struct {
+		Description  string `json:"desc"`
+		Enable       bool   `json:"enable"`
+		Radio        uint8  `json:"radio"`
+		IFValue      int32  `json:"if"`
+		Bandwidth    uint32 `json:"bandwidth,omitempty"`
+		SpreadFactor uint8  `json:"spread_factor,omitempty"`
+		Datarate     uint32 `json:"datarate,omitempty"`
+	}{
+		Description:  c.Description,
+		Enable:       c.Enable,
+		Radio:        c.Radio,
+		IFValue:      c.IFValue,
+		Bandwidth:    c.Bandwidth,
+		SpreadFactor: c.SpreadFactor,
+		Datarate:     c.Datarate,
+	})
+}
+
+// TxLUTConfig contains the configuration for the TX LUT ind
+type TxLUTConfig struct {
+	Description string `json:"desc"`
+	PAGain      int    `json:"pa_gain"`
+	MixGain     int    `json:"mix_gain"`
+	RFPower     int    `json:"rf_power"`
+	DigGain     int    `json:"dig_gain"`
+}
+
+var defaultTxLUTConfigs = []TxLUTConfig{
+	{PAGain: 0, MixGain: 8, RFPower: -6},
+	{PAGain: 0, MixGain: 10, RFPower: -3},
+	{PAGain: 0, MixGain: 12, RFPower: 0},
+	{PAGain: 1, MixGain: 8, RFPower: 3},
+	{PAGain: 1, MixGain: 10, RFPower: 6},
+	{PAGain: 1, MixGain: 12, RFPower: 10},
+	{PAGain: 1, MixGain: 13, RFPower: 11},
+	{PAGain: 2, MixGain: 9, RFPower: 12},
+	{PAGain: 1, MixGain: 15, RFPower: 13},
+	{PAGain: 2, MixGain: 10, RFPower: 14},
+	{PAGain: 2, MixGain: 11, RFPower: 16},
+	{PAGain: 3, MixGain: 9, RFPower: 20},
+	{PAGain: 3, MixGain: 10, RFPower: 23},
+	{PAGain: 3, MixGain: 11, RFPower: 25},
+	{PAGain: 3, MixGain: 12, RFPower: 26},
+	{PAGain: 3, MixGain: 14, RFPower: 27},
+}
+
+func init() {
```

Comment:
This `init` function is un-exported and not used anywhere as far as I can see.

Reply:
This sets the descriptions of `defaultTxLUTConfigs`, which is used